### PR TITLE
bucaran-fish-spec

### PR DESCRIPTION
**Pull request by bucaran**
This is my branch with working tests, I'd merge with this and then edit the details again later.

### CODE REVIEW

**Comment by bpinto** [comment]
You need to stop opening new pull requests... please, it makes it harder to follow the discussion.

**Comment by bucaran** [comment]
My apologies. Closed the last one and would like to continue the discussion here. We have green!

**Comment by bpinto** [comment]
I believe this is the right fix: https://github.com/bpinto/oh-my-fish/pull/327

**Comment by bpinto** [comment]
You shouldn't need to point to your fork anymore, as we can't merge it.

**Comment by bucaran** [comment]
Great, I will rebase and push here.

File: plugins/fish-spec/spec.eval.fish [hunk 1]
```diff
@@ -1,98 +1,29 @@
...
-        end
-      case -s --silent
-        set silent $argv[$index]
-      case -u --unit
-        # Evaluate query as a single function instead of a set.
-        set tests $query
-    end
-  end
+function spec.eval
+  set -l result 1#$status
```
Comment (bpinto) [inline plugins/fish-spec/spec.eval.fish]: You have to remove this comment. This is what is causing it to not work on my computer.

File: spec/oh-my-fish.spec.fish [hunk 2]
```diff
@@ -13,25 +13,20 @@ function describe_oh_my_fish -d "Oh-My-Fish test spec..."
...
     set -e fish_custom
     load_oh_my_fish
     expect $fish_custom --to-eq "$HOME/.oh-my-fish/custom"
   end
 
-  function it_allows_the_custom_folder_location_to_be_customized \
-    -d "It should allow modifying the location of the custom directory"
-
+  function it_allows_the_custom_folder_location_to_be_customized
+  echo hello
```
Comment (bpinto) [inline spec/oh-my-fish.spec.fish]: We don't need this `echo`.

File: plugins/fish-spec/spec/spec.spec.fish [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  # Erase functions created by the nested suite
+  spec.functions -e it_ describe_
+
+  # Restore original suite functions
+  spec.functions -re backup.
+
+  return $result
+end
+
+# spec.run $argv
```
Comment (bpinto) [inline plugins/fish-spec/spec/spec.spec.fish]: Why is this commented out?

**Comment by bpinto** [comment]
We need to change this code submission flow, it's not working very well.

Every time you rebase, you add a new change that I don't see unless I read the whole code. The flow I think you should be using instead is:

```
1. Create your branch.
2. Do your commits and push.
3. Rebase with upstream master (don't squash your commits though): git rebase upstream/master
4. Do more commits and push.
5. Rebase with upstream master (don't squash your commits though): git rebase upstream/master
6. Code is good? Review done?
  6.1 If so, squash if necessary.
  6.2 Otherwise, go back to 2.
```

**Comment by bucaran** [comment]
:+1: :+1:

**Comment by bucaran** [comment]
I am ready to push.

**Comment by bpinto** [comment]
You can push as often as you wish, I'm not sure if you have already squashed, but if not, just push the commits so I can see the new commits individually instead of checking all the PR changes, and once this PR is ready to merge, we just squash.

**Comment by bucaran** [comment]
I already squashed. Everything is looking clean.

**Comment by bpinto** [comment]
Then push. :)

**Comment by bucaran** [comment]
### Update Fish-Spec ⌁
+ Add fish eval fix See #fish-shell/pull/1892
+ New and improved output system with colors, etc.
+ Support for multiple non-nested describe blocks.
+ Remove spec.log and favor `msg` plugin and standard echo.
+ `spec.eval` no longer evaluates multiple functions, acting now as a simplified
  router between the view `spec.view` and the controller `spec.run`.
+ Add new `spec.view` to act as a hub for all test output.
+ Add report with passed/failed tests to `spec.view`
+ Now test descriptions are used directly from the function name. Description
  fields are now optional.

**Comment by bpinto** [comment]
Okay, I need to change the install script to use the branch so it correctly fetches the branch. Merging!
Thanks for all the changes, this is just amazing!

**Comment by bucaran** [comment]
Awesome.

**Pull request merged**